WooCommerce/Classes/ViewRelated/Products/Edit Product/Subscription/Trial/SubscriptionTrialViewController.swift

```diff
@@ -0,0 +1,180 @@
+                    .frame(maxWidth: .infinity, alignment: .leading)
+
+                Menu {
+                    ForEach(SubscriptionPeriod.allCases, id: \.self) { period in
+                        Button {
+                            viewModel.trialPeriod = period
+                        } label: {
+                            HStack {
+                                Text(period.descriptionSingular)
+
+                                Image(uiImage: .checkmarkStyledImage)
+                                    .resizable()
+                                    .frame(width: Layout.imageSize * scale, height: Layout.imageSize * scale)
+                                    .renderedIf(period == viewModel.trialPeriod)
+                            }
+                        }
+                    }
+                } label: {
+                    Text(viewModel.trialPeriodDescription)
+                        .bodyStyle()
+                        .multilineTextAlignment(.trailing)
+                        .frame(maxWidth: .infinity, alignment: .trailing)
+                }
+            }
+            .frame(minHeight: Layout.AdaptiveStack.height)
+            .padding(.horizontal, Layout.margin)
+
+            Divider()
+                .padding(.leading, Layout.margin)
+        }
+    }
+}
+
+private extension SubscriptionTrialView {
+    enum Layout {
+        static let margin: CGFloat = 16
+        static let verticalSpacing: CGFloat = 8
+        static let imageSize: CGFloat = 40
+
+        enum AdaptiveStack {
+            static let spacing: CGFloat = 20
+            static let height: CGFloat = 44
+        }
+    }
+
+    enum Localization {
+        static let duration = NSLocalizedString("subscriptionTrialView.duration",
+                                                       value: "Duration",
+                                                       comment: "Title for the free trial length row in add or edit subscription free trial screen.")
+        static let period = NSLocalizedString("subscriptionTrialView.period",
+                                                       value: "Period",
+                                                       comment: "Title for the free trial period row in add or edit subscription free trial screen.")
+        static let done = NSLocalizedString("subscriptionTrialView.done",
+                                                       value: "Done",
+                                            comment: "Title of the button to save subscription's free trial info.")
+        static let freeTrialSubtitle = NSLocalizedString(
+            "subscriptionTrialView.freeTrialSubtitle",
+            value: "Free trial is an optional period of time to wait before charging the first recurring payment. " +
+            "Any sign up fee will still be charged at the outset of the subscription. " +
+            "The trial period can not exceed: 90 days, 52 weeks, 24 months or 5 years.",
```

Comment:
Nit: should we remove this line to keep the subtitle short and simple. We are displaying validation error anyway so this feels redundant.

Reply:
Done in e4a5910555cc8de687ec3dd041472167094127d5